example/authentication/http/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"github.com/gazebo-web/auth/pkg/authentication"
+	"github.com/gazebo-web/auth/pkg/middleware"
+	"github.com/rs/cors"
+	"log"
+	"net/http"
+	"os"
+	"time"
+)
+
+func main() {
+	fp := os.Getenv("AUTHENTICATION_PUBLIC_KEY_FILE")
+	f, err := os.ReadFile(fp)
+	if err != nil {
+		log.Fatalln("Failed to read file:", err)
+	}
+	auth := authentication.NewTokenAuthentication(f)
+	bearer := middleware.BearerToken(auth)
+
+	srv := http.Server{
+		Addr:         ":3030",
+		Handler:      cors.AllowAll().Handler(bearer(http.HandlerFunc(handler))),
```

Comment:
Since we're trying to show users how to use the library, we should extract this and show every step of handler preparation to explain the steps we're taking.

```go
// Make the handler an `http.Handler` function
h := http.HandlerFunc(handler)

// Require a bearer token to access the handle endpoint
h = bearer(h)

// Enable CORS
h = cors.AllowAll().Handler(h)

	srv := http.Server{
		Addr:         ":3030",
		Handler:      cors.AllowAll().Handler(bearer(http.HandlerFunc(handler))),
```